cassovary-core/src/main/scala/com/twitter/cassovary/graph/MemoryMappedDirectedGraph.scala

```diff
@@ -122,190 +159,164 @@ object MemoryMappedDirectedGraph {
-    */
-  private def writeHeaderAndDegrees(nodeCount: Int, outDegree: Int => Int, inDegree: Int => Int, out: DataOutput): Unit = {
-    out.writeInt(0)
-    out.writeInt(nodeCount)
-    // The out-neighbor data starts after the initial 8 bytes, n+1 Longs for out-neighbors and n+1 Longs for in-neighbors
-    var outboundOffset = 8L + 8L * (nodeCount + 1) * 2
-    for (i <- 0 until nodeCount) {
-      out.writeLong(outboundOffset)
-      outboundOffset += 4 * outDegree(i)
-    }
-    out.writeLong(outboundOffset) // Needed to compute outdegree of node n-1
+    var edgesReadCount = 0L // Only needed for logging
+    var maxNodeId = 0
 
-    // The inbound data starts immediately after the outbound data
-    var inboundOffset = outboundOffset
-    for (i <- 0 until nodeCount) {
-      out.writeLong(inboundOffset)
-      inboundOffset += 4 * inDegree(i)
-    }
-    out.writeLong(inboundOffset) // Needed to compute indegree of node n-1
-  }
+    log("started reading graph at " + new Date())
+    forEachEdge(edgeListFile) { (id1, id2) =>
+      maxNodeId = math.max(maxNodeId, math.max(id1, id2))
+      // Increase the number of temp files if needed
+      while (maxNodeId >= tempFilesById1.size * nodesPerChunk) {
+        createAndStoreNewTemporaryFile(tempFilesById1, outStreamsById1)
+        createAndStoreNewTemporaryFile(tempFilesById2, outStreamsById2)
+      }
 
-  /** Converts a graph to binary format.  The input is a file containing lines  of the form
-    * "<id1> <id2>", and this method will throw an IOException if any non-blank line of the file doesn't have this form.
-    * The graph in binary format is written to the given file.
-    * TODO: This method is currently rather slow (~5 hours on a ~800M edge graph), and would likely
-    * be faster if we used a memory mapped file rather than a RandomAccessFile.  The method
-    * sortedEdgeFilesToGraph is currently faster (~1 hour on a ~1B edge graph).
-    * */
-  def edgeFileToGraph(edgeListFile: File, graphFile: File): Unit = {
-    val (outDegrees, inDegrees) = accumulateOutAndInDegrees(edgeListFile)
-    System.err.println("finished reading degrees at " + new Date())
-    val nodeCount = outDegrees.size()
-    val out = new RandomAccessFile(graphFile, "rw")
-    def outDegree(id: Int): Int = outDegrees.get(id)
-    def inDegree(id: Int): Int = inDegrees.get(id)
-    writeHeaderAndDegrees(nodeCount, outDegree, inDegree, out)
-
-    // outboundOffsets(i) stores the offset where the next out-neighbor of node i should be written
-    val outboundOffsets = new Array[Long](nodeCount)
-    //The outneighbor data starts after the initial 8 bytes, n+1 Longs for outneighbors and n+1 Longs for in-neighbors
-    var cumulativeOffset = 8L + 8L * (nodeCount + 1) * 2
-    for (i <- 0 until nodeCount) {
-      outboundOffsets(i) = cumulativeOffset
-      cumulativeOffset += 4 * outDegree(i)
+      val outStreamById1 = outStreamsById1(id1 / nodesPerChunk)
+      outStreamById1.writeInt(id1)
```

Comment:
This part is probably confusing to read because there are both fileChannels and BufferedOutputStreams around.  outStreamById1 writes to a BufferedOutputStream, so the writeInt call just appends 4 bytes to a buffer (array), and is the data is ultimately written in 8KB chunks.  Do you think I should add a comment saying outStreamById1 is buffered?